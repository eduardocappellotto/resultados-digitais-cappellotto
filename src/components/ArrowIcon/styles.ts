import styled, { css } from 'styled-components'

type WrapperProps = {
  $direction: 'left' | 'up' | 'right' | 'down'
}

const wrapperModifiers = {
  left: () => css`
    transform: rotate(-180deg);
  `,
  up: () => css`
    transform: rotate(270deg);
  `,
  right: () => css`
    transform: rotate(0deg);
  `,
  down: () => css`
    transform: rotate(90deg);
  `
}

export const Wrapper = styled.div<WrapperProps>`
  ${({ $direction }) => css`
    display: inline-flex;
    justify-content: center;
    align-items: center;

    ${!!$direction && wrapperModifiers[$direction]()};
  `}
`
