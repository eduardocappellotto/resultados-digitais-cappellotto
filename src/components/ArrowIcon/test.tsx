import { render, screen } from "@testing-library/react";

import { ArrowIcon } from ".";

describe("<ArrowIcon />", () => {
  it("should render the right arrow icon as default", () => {
    render(<ArrowIcon />);

    expect(screen.getByRole("img", { name: /arrow icon/i })).toHaveAttribute(
      "width",
      "16"
    );
    expect(
      screen.getByRole("img", { name: /arrow icon/i }).parentElement
    ).toHaveStyle({
      transform: "rotate(0deg)",
    });
  });

  it("should render the left arrow icon", () => {
    render(<ArrowIcon direction="left" />);

    expect(
      screen.getByRole("img", { name: /arrow icon/i }).parentElement
    ).toHaveStyle({
      transform: "rotate(-180deg)",
    });
  });

  it("should render the up arrow icon", () => {
    render(<ArrowIcon direction="up" />);

    expect(
      screen.getByRole("img", { name: /arrow icon/i }).parentElement
    ).toHaveStyle({
      transform: "rotate(270deg)",
    });
  });

  it("should render the down arrow icon", () => {
    render(<ArrowIcon direction="down" />);

    expect(
      screen.getByRole("img", { name: /arrow icon/i }).parentElement
    ).toHaveStyle({
      transform: "rotate(90deg)",
    });
  });
});
