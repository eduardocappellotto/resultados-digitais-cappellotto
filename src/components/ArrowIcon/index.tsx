import * as S from "./styles";

export type ArrowIconProps = {
  direction?: "left" | "up" | "right" | "down";
  size?: number;
};

export function ArrowIcon({ direction = "right", size = 16 }: ArrowIconProps) {
  return (
    <S.Wrapper $direction={direction}>
      <svg
        xmlns="http://www.w3.org/2000/svg"
        width={size}
        fill="none"
        viewBox="0 0 25 24"
        role="img"
        aria-label="Arrow icon"
      >
        <path
          fill="currentColor"
          d="m16.385 8.143 3.974 4.013a.49.49 0 0 1 0 .688l-3.974 4.013a.48.48 0 0 1-.822-.344v-2.878H4.982a.484.484 0 0 1-.482-.487v-1.296c0-.269.216-.487.482-.487h10.581V8.487a.48.48 0 0 1 .822-.344Z"
        />
      </svg>
    </S.Wrapper>
  );
}
