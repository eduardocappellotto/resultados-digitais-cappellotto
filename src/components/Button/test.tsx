import { fireEvent, screen } from "@testing-library/react";

import { Button } from ".";
import { renderWithTheme } from "@/utils/tests/helpers";
import { ArrowIcon } from "../ArrowIcon";

describe("<Button />", () => {
  it("should render the button", () => {
    const { container } = renderWithTheme(<Button>Button</Button>);

    expect(screen.getByRole("button")).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });

  it("should render icon on the left of the button", () => {
    renderWithTheme(
      <Button icon={<ArrowIcon />} iconPosition="left">
        Button
      </Button>
    );

    expect(screen.getByRole("button")).toHaveStyle({
      flexDirection: "row-reverse",
    });
  });

  it("should call method when button is clicked", () => {
    const onClick = jest.fn();
    renderWithTheme(<Button onClick={onClick}>Button</Button>);

    fireEvent.click(screen.getAllByRole("button")[0]);

    expect(onClick).toHaveBeenCalled();
  });

  it("shouldn't call method when button is clicked and disabled", () => {
    const onClick = jest.fn();
    renderWithTheme(
      <Button onClick={onClick} disabled>
        Button
      </Button>
    );

    fireEvent.click(screen.getAllByRole("button")[0]);

    expect(onClick).not.toHaveBeenCalled();
  });
});
