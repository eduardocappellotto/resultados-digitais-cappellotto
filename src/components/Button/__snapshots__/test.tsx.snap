// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render the button 1`] = `
.c0 {
  width: 100%;
  height: 4.8rem;
  display: flex;
  gap: 0.8rem;
  justify-content: center;
  align-items: center;
  border: 1px solid;
  background-color: #F2BF4E;
  line-height: 24px;
  font-size: 1.6rem;
  font-weight: 700;
  text-transform: uppercase;
  color: #100C27;
  letter-spacing: 0.2px;
  cursor: pointer;
  -webkit-box-shadow: 0px 6px 1px -2px #000000;
  -moz-box-shadow: 0px 6px 1px -2px #000000;
  box-shadow: 0px 6px 1px -2px #000000;
  transition: 0.1s;
  flex-direction: row;
}

.c0:hover {
  background-color: #F8DCA0;
  border-color: transparent;
}

.c0[disabled] {
  background-color: #E1E4E8;
  color: #97A1AC;
  pointer-events: none;
  border-color: transparent;
  -webkit-box-shadow: 0px 6px 1px -2px #97A1AC;
  -moz-box-shadow: 0px 6px 1px -2px #97A1AC;
  box-shadow: 0px 6px 1px -2px #97A1AC;
}

<button
  class="c0"
>
  <span>
    Button
  </span>
</button>
`;
